src/megaclient.cpp

```diff
@@ -7601,10 +7627,221 @@ void MegaClient::fetchnodes()
         }
 #endif
 
+        fetchkeys();
         reqs.add(new CommandFetchNodes(this));
     }
 }
 
+void MegaClient::fetchkeys()
+{
+#ifdef ENABLE_CHAT
+
+    fetchingkeys = true;
+
+    User *u = finduser(me, 1);
+
+    int creqtag = reqtag;
+    reqtag = 0;
+    reqs.add(new CommandPubKeyRequest(this, u));    // public RSA
+    reqtag = creqtag;
+
+    getua(u, "*keyring", 0);   // private Cu25519 & private Ed25519
+    getua(u, "+puEd255", 0);
+    getua(u, "+puCu255", 0);
+    getua(u, "+sigCu255", 0);
+    getua(u, "+sigPubk", 0); // it triggers MegaClient::initializekeys()
```

Comment:
It seems that there are several code paths in `CommandGetUA::procresult()` that could keep `fetchingkeys` to `true` forever.